zookeeper-server/src/main/java/org/apache/zookeeper/version/util/VerGen.java

```diff
@@ -119,7 +119,7 @@ public static void generateFile(File outputDir, Version version, String rev, Str
             w.write("public class " + VERSION_CLASS_NAME + " implements " + PACKAGE_NAME + ".Info {\n");
             w.write("    public static void main(String[] args) {\n");
             w.write("        final String VER_STRING = MAJOR + \".\" + MINOR + \".\" + MICRO +");
-            w.write("            (QUALIFIER == null ? \"\" : \"-\" + QUALIFIER)  + \" \" +");
+            w.write("            (QUALIFIER.isEmpty() ? \"\" : \"-\" + QUALIFIER)  + \" \" +");
```

Comment:
What happens if it is null?

Reply:
In line 83, it is set to empty string, if it is null - ` w.write("    String QUALIFIER=" + (version.qualifier == null ? "\"\"" : "\"" + version.qualifier + "\"") + ";\n");` which the VersionInfoMain implements.

Reply:
Could do:

```suggestion
            w.write("            (QUALIFIER == null || QUALIFIER.isEmpty() ? \"\" : \"-\" + QUALIFIER)  + \" \" +");
```

Reply:
Ya, that was the logic in the first commit, i updated it to QUALIFIER.isEmpty() in the second update. @ctubbsii  I'll update the PR, thank you for the review.

Reply:
@eolivelli and @ctubbsii I have made the change to include both the null check and the empty string check. Please let me know if this is alright. Thanks!

Reply:
Thanks, will resolve the comments then.